modules/travel_pay/app/services/travel_pay/token_service.rb

```diff
@@ -1,21 +1,52 @@
 # frozen_string_literal: true
 
+require_relative '../../models/travel_pay/travel_pay_cache'
+
 module TravelPay
   class TokenService
     #
     # returns a hash containing the veis_token & btsss_token
     #
     def get_tokens(current_user)
-      veis_token = token_client.request_veis_token
-      btsss_token = token_client.request_btsss_token(veis_token, current_user)
+      cached = cached_by_account_uuid(current_user.account_uuid)
+      if cached
```

Comment:
This doesn't appear to check the access token expiration claim, so if it's expired, it'll just keep pulling an old token from the cache. It should be up to the token service instance to check the exp date and refresh the token.

Reply:
That might be another reason to store the tokens independently. The BTSSS token has an expiration date of 1hr, the veis token is way less, like 15 mins I think (I'll verify). So that means we're not taking advantage of the BTSSS token's longer exp date when we add the refresh logic, and so we're refreshing it more often than we need to. With the strong feelings of the API about calling the token every time we make a request, this is something to consider.

Reply:
Good catch! I totally missed that check! - Although I set the ttl to 30 minutes, so after 30 minutes they won't exist in the DB to be pulled any more.

(On a related note, do we know how long the tokens are good for once they're retrieved? I marked the ttl for 30 minutes, based on how long the user's session typically is, with the expectation that we'd make the redis TTL slightly shorter than the expiration time on the tokens themselves.)

Reply:
BTSSS Token: 60 mins
VEIS Token: 60 mins